catalyst/core/callbacks/checkpoint.py

```diff
@@ -107,22 +107,26 @@ def get_checkpoint_suffix(self, checkpoint: dict) -> str:
     @staticmethod
     def load_checkpoint(*, filename, state: _State):
         if os.path.isfile(filename):
-            print(f"=> loading checkpoint {filename}")
             checkpoint = utils.load_checkpoint(filename)
-
-            state.epoch = checkpoint["epoch"]
-
-            utils.unpack_checkpoint(
-                checkpoint,
-                model=state.model,
-                criterion=state.criterion,
-                optimizer=state.optimizer,
-                scheduler=state.scheduler
-            )
-
-            print(
-                f"loaded checkpoint {filename} (epoch {checkpoint['epoch']})"
-            )
+            if checkpoint["epoch"] > state.epoch:
```

Comment:
what is this for?

Reply:
When you resume on stage1, on stage2 you will resume again. This `if` helps to solve the problem.

Reply:
@AndreySheka could you please add a comment about this in the code?
or could we add some additional logic to check if we need to resume checkpoint loading again?
maybe add `self.resume = None` in the end of each stage?

this if looks really unclear and hacky for me, that the case :)

Reply:
I remove this hack, now checkpoint callback clear `state.resume` after loading checkpoint and `resume` field added into migration list of `runner`